dkc/core/models/file.py

```diff
@@ -64,5 +64,13 @@ def clean(self) -> None:
 
 @receiver(models.signals.pre_save, sender=File)
 def _file_pre_save(sender, instance, *args, **kwargs):
-    # TODO this is where we might handle quotas
     instance.size = instance.blob.size
+
+    if not instance.pk:  # this is a new file; reserve quota
+        instance.folder.increment_quota(instance.size)
```

Comment:
This is a temporary place for incrementing the quota. Ultimately we'll want to do that when they first initiate an upload, before any bytes get stored anywhere.